controllers/datadogagent/merger/cilium.go

```diff
@@ -0,0 +1,591 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package merger
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/component"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/dependencies"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object"
+	cilium "github.com/DataDog/datadog-operator/pkg/cilium/v1"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
+)
+
+// CiliumPolicyManager is used to manage cilium policy resources.
+type CiliumPolicyManager interface {
+	CreateCiliumPolicy(name, namespace string, policySpecs []cilium.NetworkPolicySpec) error
+	BuildCiliumPolicy(dda metav1.Object, component v2alpha1.ComponentName) error
+	SetupCiliumManager(site, ddURL string, hostNetwork bool, dnsSelectorEndpoints []metav1.LabelSelector)
+}
+
+// NewCiliumPolicyManager returns a new CiliumPolicyManager instance
+func NewCiliumPolicyManager(store dependencies.StoreClient) CiliumPolicyManager {
+	manager := &ciliumPolicyManagerImpl{
+		store: store,
+	}
+	return manager
+}
+
+// ciliumPolicyManagerImpl is used to manage cilium policy resources.
+type ciliumPolicyManagerImpl struct {
+	store dependencies.StoreClient
+
+	site                 string
+	ddURL                string
+	hostNetwork          bool
+	dnsSelectorEndpoints []metav1.LabelSelector
+}
+
+func (m *ciliumPolicyManagerImpl) SetupCiliumManager(site, ddURL string, hostNetwork bool, dnsSelectorEndpoints []metav1.LabelSelector) {
+	m.site = site
+	m.ddURL = ddURL
+	m.hostNetwork = hostNetwork
+	m.dnsSelectorEndpoints = dnsSelectorEndpoints
+}
+
+// AddCiliumPolicies creates or updates multiple cilium network policies
+func (m *ciliumPolicyManagerImpl) CreateCiliumPolicy(name, namespace string, policySpecs []cilium.NetworkPolicySpec) error {
```

Comment:
nit the name in the comment doesn't match function name. (perhaps use AddCiliumPolicy to be consistent with the kubernetes network policy version?)